lr_face/experiments.py

```diff
@@ -35,48 +37,57 @@ def __str__(self):
         return '_'.join(map(str, [
             self.scorer,
             self.calibrator,
-            data_str,
+            # data_str,
             params_str
         ])).replace(':', '-')  # Windows forbids ':'
 
     def get_calibration_and_test_pairs(self) -> Tuple[
-        List[FacePair],
-        List[FacePair]
+        Dict[Tuple, List[FacePair]],
+        Dict[Tuple, List[FacePair]]
     ]:
-        datasets = self.data_config['datasets']
-
-        # If `datasets` is a single `Dataset` instance, split its images by
-        # identity into two disjoint sets and make pairs out of them.
-        if isinstance(datasets, Dataset):
-            test_size = self.data_config['fraction_test']
-            if datasets.pairs:
-                calibration_pairs, test_pairs = train_test_split(
-                    datasets.pairs,
-                    test_size=test_size,
-                    stratify=[p.same_identity for p in datasets.pairs])
-            else:
-                calibration_pairs, test_pairs = map(
-                    make_pairs,
-                    split_by_identity(datasets, test_size)
-                )
-            return calibration_pairs, test_pairs
-
-        # If `datasets` is already a tuple of `Dataset` instances, make pairs
-        # for each individual dataset and return those.
-        if isinstance(datasets, tuple) \
-                and len(datasets) == 2 \
-                and all(isinstance(x, Dataset) for x in datasets):
-            calibration_pairs = datasets[0].pairs
-            if not calibration_pairs:
-                calibration_pairs = make_pairs(datasets[0])
-            test_pairs = datasets[1].pairs
-            if not test_pairs:
-                test_pairs = make_pairs(datasets[1])
-            return calibration_pairs, test_pairs
-
-        # In all other cases something was misconfigured, so raise an error.
-        raise ValueError(
-            f'Could not create calibration and test data from {str(datasets)}')
+        assert isinstance(self.data_config['calibration'], tuple)
+        assert isinstance(self.data_config['test'], tuple)
+
+        # get all images
+        calibration_images = []
+        for dataset in self.data_config['calibration']:
+            calibration_images += dataset.images
+
+        # filter the images per category
```

Comment:
Het is voor mij niet helemaal duidelijk was nu een categorie is, is dit bijvoorbeeld: "geen hoofddeksel" of een combinatie van bijvoorbeeld "geen hoofddeksel, matige qualiteit, pitch frontal". Want ik heb het idee dat dit het eerste is, maar dat we eigenlijk het tweede zoude willen voor de spraak-methode-callibratie.

Reply:
het is wel het tweede, een tuple van alle combinaties